docs/proposals/scheduling/20220510-load-aware-scheduling.md

```diff
@@ -0,0 +1,137 @@
+
+#### Story 2
+
+Workloads in a co-located cluster have different resource requirements. Typical CPU-bound workloads expect to use more CPU, while other types of workloads may use more memory. If only pay attention to the utilization threshold of a certain resource, it may cause the utilization of another resource to be low or high, which will lead to the insufficient use of resources in the global dimension.
+
+#### Story 3
+
+Koordinator defines NodeMetric CRD to describe the resource usage of nodes and is regularly updated by koordlet. However, if there are many Pods scheduled to cold nodes (that is, nodes with low resource utilization) during the update cycle, when these Pods start running, the resource utilization of these nodes may exceed the expected threshold. As a result, the runtime quality of these pods is not as good as expected.
+
+#### Story 4
+
+The koordlet may not be able to report the latest resource usage due to node exceptions. Such nodes should be avoided during scheduling to prevent unexpected exceptions.
+
+### Notes/Constraints
+
+### Implementation Details
+
+![image](/docs/images/load-aware-scheduling-arch.svg)
+
+The scheduling plugin filters abnormal nodes and scores them according to resource usage. This scheduling plugin extends the Filter/Score/Reserve/Unreserve extension points defined in the Kubernetes scheduling framework.
+
+#### Filter unhealthy nodes
+
+By default, abnormal nodes are filtered, and users can decide whether to enable or not by configuring as needed.
+
+- Filter nodes where koordlet fails to update NodeMetric. If the configuration enables, the plugin will exclude nodes with *nodeMetrics.status.updateTime >= nodeMetricUpdateMaxWindowInSeconds*
+- Filter nodes by utilization thresholds. If the configuration enables, the plugin will exclude nodes with *latestUsageUtilization >= utilizationThreshold*. In the filtering phase, only the resource utilization is obtained from the latest NodeMetric, and the resource usage of the allocated but not yet counted Pods does not participate in the calculation, so as to allocate resources to the newly created Pods and avoid scheduling failure due to unreasonable estimates
+
+
+#### Score algorithm
+
+The core logic of the scoring algorithm is to select the node with the smallest resource usage. However, considering the delay of resource usage reporting and the delay of Pod startup time, the resource requests of the Pods that have been scheduled and the Pods currently being scheduled within the time window will also be estimated, and the estimated values will be involved in the calculation.
+
+The following is the algorithm:
+1. Get the total amount of resources of the current node as *T*
+2. Get the resource usage of the current node from NodeMetric as *A*
+3. Estimate the resource usage of the current Pod as *B*
+    1. If Limit is declared in PodSpec, then *B = Limit*
+    2. else *B = Request * scalingFactor*
```

Comment:
maybe *B = min(Request * scalingFactor, Limit)* a better solution when scalingFactor>1.0 ? anyway it depends on the default literal value of scalingFactor.

Reply:
It is reasonable to take the minimum value here. Thanks.